pyproject.toml

```diff
@@ -26,6 +26,11 @@ classifiers = [
 ]
 dependencies = [
   "ipykernel>=6.29.5",
+  "mkdocs>=1.6.1",
+  "mkdocs-bootstrap4>=0.1.5",
+  "mkdocs-material>=9.6.7",
+  "mkdocs-spellcheck[all]>=1.1.0",
+  "mkdocstrings-python>=1.11.1",
```

Comment:
These are in the wrong block I believe, they should just be on the `tool.hatch.env.docs` part, and installed only when in the `docs` environment.  Otherwise in order to install the package for use you'll need all this